src/main/java/frc/robot/joystickManager/JoystickManager.java

```diff
@@ -0,0 +1,73 @@
+package frc.robot.joystickManager;
+
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.Robot;
+import frc.joysticks.JoystickPorts;
+import frc.joysticks.SmartJoystick;
+
+import frc.joysticks.JoystickPorts.*;
+
+public class JoystickManager {
+
+	private SmartJoystick[] joysticks;
+
+	private final Robot robot;
+
+	public JoystickManager(Robot robot) {
+		joysticks = new SmartJoystick[6];
+
+		int currentPort = 0;
+
+		for (SmartJoystick joystick : joysticks) {
+			SendableChooser<JoystickState> stateChooser = new SendableChooser<>();
+			addOptions(stateChooser, intToPort(currentPort));
+			currentPort++;
+		}
+
+		this.robot = robot;
+	}
+
+	public void setJoystickState(JoystickPorts port, JoystickState joystickState) {
+		if (portToJoystick(port) != null) {
+			portToJoystick(port).setState(joystickState);
+		} else if (portToJoystick(port) == null) {
+			joysticks[port.getPort()] = new SmartJoystick(port, joystickState);
+			JoystickBindings.configureBindings(joysticks[port.getPort()], robot);
+		}
+	}
+
+	private SmartJoystick portToJoystick(JoystickPorts port) {
```

Comment:
is this still needed now that its just array access?
in general is there any need for port to be an enum and not an int saving the int to port function as well?

Reply:
you have limited number of ports

Reply:
fine but the first point still stands